lib/src/hash_sink.dart

```diff
@@ -121,12 +121,14 @@ abstract class HashSink implements Sink<List<int>> {
       _pendingData.add(0);
     }
 
-    var lengthInBits = _lengthInBytes * bitsPerByte;
-    if (lengthInBits > maxUint64) {
+    const maxMessageLengthInBytes = ((1 << (64 - bitsPerByte)) - 1);
```

Comment:
Nit: the outer parentheses are unnecessary.

I'd make this a top-level variable, not a local.

This bit munging is pretty opaque. Can you just make this a number literal instead?